dftimewolf/lib/processors/gcp_cloud_resource_tree.py

```diff
@@ -0,0 +1,1199 @@
+    """Return a string representation of the resource tree."""
+    output = '\n'
+    dashes = '-' * 200
+
+    # Draw table header
+    output = output + dashes + '\n'
+    # pylint: disable=line-too-long
+    output = output + '{:<20s}|{:<18s}|{:<20s}|{:<25s}|{:<18s}|{:<20s}|{:<25s}|{:<18s}|{:<100s}\n'.format(
+        'Resource ID', 'Resource Type', 'Creation TimeStamp', 'Created By',
+        'Creator IP Addr', 'Deletion Timestamp', 'Deleted By',
+        'Deleter IP Addr', 'Tree')
+    output = output + dashes + '\n'
+
+    result = self.GenerateTree()
+    for i in result:
+      resource: Optional[Resource] = i.get('resource_object')
+      if resource:
+        output = output + \
+            ('{:<20s}|{:<18s}|{:<20s}|{:<25s}|{:<18s}|{:<20s}|{:<25s}|{:<18s}|{:<100s}\n'.format( resource.id if ('-' not in resource.id) else "" , resource.type, resource.creation_timestamp.astimezone(timezone.utc).strftime("%Y-%m-%d %H:%M:%S") if resource.creation_timestamp else "",
+            resource.created_by, resource.creator_ip_address,  resource.deletion_timestamp.astimezone(timezone.utc).strftime("%Y-%m-%d %H:%M:%S") if resource.deletion_timestamp else "", resource.deleted_by, resource.deleter_ip_address,  i.get('graph')))
+    output = output + dashes + '\n'
+
+    return output
+
+
+class GCPCloudResourceTree(module.BaseModule):
+  """GCP Cloud Resource Tree Creator.
+
+  input: None, takes input from parameters only.
+  output: Temp file with a dump of the resource tree
+  """
+
+  def __init__(self,
+               state: DFTimewolfState,
+               name: Optional[str] = None,
+               critical: bool = True) -> None:
+    """Initialize the Cloud Resource Tree Processor.
+
+    Args:
+      state: recipe state.
+      name: The module's runtime name.
+      critical: True if the module is critical, which causes
+          the entire recipe to fail if the module encounters an error.
+    """
+    super(GCPCloudResourceTree, self).__init__(state,
+                                               name=name,
+                                               critical=critical)
+
+    self._project_id = ''
+    self._resource_name = ''
+    self._resource_type = ''
+    self._mode: OperatingMode = OperatingMode.ONLINE
+    self._start_date = ''
+    self._end_date = ''
+    self._period_covered_by_retrieved_logs: Dict[str, datetime] = {}
+    self._resources_dict: Dict[str, Resource] = {}
+
+  # pylint: disable=arguments-differ
+  def SetUp(self, project_id: str, resource_name: str, resource_type: str,
+            mode: str) -> None:
+    """Set up the resource we want to build the tree for.
+
+    Args:
+      project_id: Project id where the resources are located.
+      resource_name: Resource name.
+      resource_type: Resource type (currently supported types: gce_instance,
+        gce_disk, gce_image, gce_machine_image, gce_instance_template,
+        gce_snapshot)
+      mode: operational mode: online or offline
```

Comment:
get rid of the second ":", maybe "mode: The operational mode, which is either online or offline."